[Card Game 3] Карта + Рука

SenX51 opened the pull request:
**Выполнил:**
 > Семенов А.В.

**Задача:**
 > Реализовать компонент карточки (картинка фоном, фрагменты с цифрами (сила и тд) и реализация контейнера (руки) персонажа с просмотром карт. Карты располагать веером, при наведении показывать карту подробнее и как бы “вытягивать” ее из руки выше

**Комментарии:**
 > Прошу прощения за задержку, сломался ПК, 2 недели ждал процессор по почте

**Демонстрация:**
[Видео](https://github.com/user-attachments/assets/bbb1691a-a8c0-4000-b472-55352923fd45)
![image](https://github.com/user-attachments/assets/273ba42e-efcd-4737-84df-80f1eef5ff9f)
![image](https://github.com/user-attachments/assets/292d421c-c180-4caf-9aa7-61e7e4b7ec61)

SenX51 committed:
Created: Card,  Deck

Created:
> Card component and store
> Deck component mock and basic store structure
> Established File Structure

TODO:
> Fix Cards being clipped
> Scale down Cards to fit Deck height
> Implement Deck store functionality
> Implement Game Logic

SenX51 committed:
Syncronization commit

commit to syncronize code between devices

SenX51 committed:
Card behaviour in deck

Fixed:
 > Cards correctly respond to hover
 > Cards automatically create description

reznikovkg commented on vue/src/components/card/CardComponent.vue:
```diff
@@ -0,0 +1,205 @@
+<template>
+	<div class="card__wrapper" :style="applyTransform" @mouseover="isHovered = true" @mouseleave="isHovered = false"
```
2 пробела на отступ
Выровнить
Перенести атрибуты
Стрелочные функции

reznikovkg commented on vue/src/components/card/CardComponent.vue:
```diff
@@ -0,0 +1,205 @@
+<template>
+	<div class="card__wrapper" :style="applyTransform" @mouseover="isHovered = true" @mouseleave="isHovered = false"
+		@click="cardClick">
+		<div class="card__div hCenter">
```
Писать классы по БЭМу kebab-case

reznikovkg commented on vue/src/components/card/CardComponent.vue:
```diff
@@ -0,0 +1,205 @@
+					<p> {{ this.card.title }} </p>
+				</div>
+				<div class="card-description__description">
+					<p> {{ this.cardDescription }}</p>
```
this убрать

reznikovkg commented on vue/src/components/card/CardComponent.vue:
```diff
@@ -0,0 +1,205 @@
+	},
+	computed: {
+		applyTransform() {
+			if (this.isHovered)
```
Вынести в логику стола

SenX51 committed:
Fix card responsibility

FIX:
 > Логика выкладывания карт перенесена в компонент руки
 > Константы типов карт прописаны в верхнем регистре
 > Стайлинг, нейминг, стрелочные функции
 > Появление скролла/смещение страницы при проигрывании анимации карт
ADD:
 > Расстояние между картами уменьшается пропорционально количеству карт
 > Задний фон для изображения на карте

SenX51 commented:
@reznikovkg внес правки в код

reznikovkg commented on vue/src/components/card/CardComponent.vue:
```diff
@@ -0,0 +1,197 @@
+  >
+    <div class="card__content center-horizontal">
+      <div class="card__cost center-horizontal center-vertical">
+        <p>{{ (() => card.cost)() }}</p>
```
Нельзя такое делать, просто  {{ card.cost }}

reznikovkg commented on vue/src/components/card/CardComponent.vue:
```diff
@@ -0,0 +1,197 @@
+      this.$emit('onMouseOver', isHovered);
+    },
+
+		generateDescription() {
```
Поехало

reznikovkg commented on vue/src/components/deck/DeckComponent.vue:
```diff
@@ -0,0 +1,113 @@
+      <CardComponent 
+        v-for="(card, index) in getHand" 
+        :key="(() => card.id)()"
+        :card="(() => card)()"
```
Просто :card="card" и везде так. Зачем функции тут? Функции только для событий

SenX51 committed:
Unanonymized variables

reznikovkg commented on vue/src/components/card/CardComponent.vue:
```diff
@@ -0,0 +1,196 @@
+		text-wrap: nowrap;
+	}
+
+	&__image {
```
Тут поехало

reznikovkg commented on vue/src/components/card/CardComponent.vue:
```diff
@@ -0,0 +1,196 @@
+    margin-top: 10px;
+    border-radius: 15px;
+    
+		&-img {
```
Вообще сильно поехало

reznikovkg commented on vue/src/components/deck/DeckComponent.vue:
```diff
@@ -0,0 +1,113 @@
+
+    applyTransform(index) {
+      const isHovered = this.hoveredIndex == index;
+      if (isHovered)
```
Проверьте везде по выравниванию

reznikovkg commented on vue/src/store/cards.js:
```diff
@@ -0,0 +1,81 @@
+import Vuex from "vuex";
+import cardEffect from "@/components/common/cardEffect";
+
+Vue.use(Vuex);
```
Не нужно создавать экземпляр стора везде несколько раз. Это должно быть только раз в index, остальные подключаются как подмодули

SenX51 committed:
Fixed auto-formatter mess ups

reznikovkg commented:
Сдано
